src/solvers/dgsem_tree/containers.jl

```diff
@@ -28,9 +28,11 @@ function reinitialize_containers!(mesh::TreeMesh, equations, dg::DGSEM, cache)
     init_boundaries!(boundaries, elements, mesh)
 
     # re-initialize mortars container
-    @unpack mortars = cache
-    resize!(mortars, count_required_mortars(mesh, leaf_cell_ids))
-    init_mortars!(mortars, elements, mesh)
+    if hasproperty(cache, :mortars) # cache_parabolic does not carry mortars
```

Comment:
OK to leave it like this (for now), but do you think it would make sense to have `is_cache_parabolic(cache)` to detect this more canonically? If it is just required here, it is probably not worth it though...

Reply:
Maybe this is something we can consider in step 3 of https://github.com/trixi-framework/Trixi.jl/issues/1674 (cleaning up `cache_parabolic`)?

Reply:
One thing I mentioned in one of the discussions above is that we could also avoid the `if` by introducing a specialization for `reinitialize_containers!(mesh::TreeMesh, equations, dg::DGSEM, cache::cache_parabolic)`

Reply:
Yes, but there is no `cache_parabolic` type, so you'd have to come up with a similar work around

Reply:
True. Also the specialization `equations::AbstractEquations` does not work. The simplest thing would be implementing `reinitialize_containers_parabolic!` to avoid the `if`.